libraries/core-react/src/hooks/useOutsideClick.ts

```diff
@@ -2,10 +2,16 @@ import { useEffect, MutableRefObject } from 'react'
 
 export const useOutsideClick = (
   ref: MutableRefObject<HTMLElement>,
+  anchorEl: HTMLElement,
```

Comment:
Do we really need this? I would like to avoid this complexity as hooks are intended to solve simple problems. Can this be solved differently? Maybe inside the individual components?

Reply:
I sat with this problem for a while, and had to add this check because on first _click_ (to open menu) it closed before it had the chance to open. Had to click the button twice to open the menu after first render. 

I think it's because Popper uses anchorEl to calculate the positioning.

Reply:
Not sure if I get what you mean by adding a hook for anchorEl?